assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/ComplianceFeatureUtils.java

```diff
@@ -96,4 +98,85 @@ public static String mapEnvironmentVariable(String value) {
 
         return value;
     }
+
+    /**
+     * Reads in a properties file, replaces references to environment
+     * variables, creates a (partially initialized) {@code AntennaContext}, and
+     * returns the result as a {@code MappedConfigurationData} object. This
+     * method bridges between the limited configuration of the compliance tool
+     * and the configuration of Antenna. Although the compliance tool does not
+     * need the full configuration settings supported by Antenna, it uses some
+     * central services that need to be configured (e.g. the central HTTP
+     * client). To make this possible, parts of the configuration for the
+     * compliance tool are used to create an Antenna context, from which those
+     * service objects can be obtained.
+     *
+     * @param propertiesFile the properties file with the configuration of the
+     *                       compliance tool
+     * @return the resulting {@code MappedConfigurationData} object
+     */
+    public static MappedConfigurationData createMappedConfigurationData(File propertiesFile) {
+        Map<String, String> properties = mapPropertiesFile(propertiesFile);
+
+        ToolConfiguration.ConfigurationBuilder configurationBuilder =
+                new ToolConfiguration.ConfigurationBuilder();
+        if (Boolean.parseBoolean(properties.get("proxyUse"))) {
+            configurationBuilder.setProxyHost(properties.get("proxyHost"))
+                    .setProxyPort(Integer.parseInt(properties.get("proxyPort")));
+        }
+        AntennaContext context = new AntennaContext.ContextBuilder()
+                .setToolConfiguration(configurationBuilder.buildConfiguration())
+                .buildContext();
+        return new MappedConfigurationData(properties, context);
```

Comment:
Honestly, I do not like this solution myself. I think the basic problem is that the compliance tool is a tool on its own, but should use some core functionality provided by Antenna. This is not only limited to the HTTP client and the object mapper but includes things like logging and the handling of certain default command line options. So there needs to be a balance between code reuse and duplication.

The approach to reuse the whole context is obviously not satisfying as it has too many other dependencies. I will try to come up with another solution; but it won't be easy to find the right balance.

Reply:
From my point of view, there is no big issue since the compliance tool is also an Antenna based tooling. Each frontend can of course use the basic infrastructure provided by Antenna.

But of course having a very small need and dragging in a whole framework typically does not feel good with good reason, so I would appreciate if there is a simpler solution, but I would also accept, if you say this is conceptually the right way to go.

Reply:
This part has been reworked. The AntennaContext is no longer used by the compliance tool. Rather, the logic to create shared service objects has been extracted into a new ServiceFactory class. This one is now used by the compliance tool, which is a significant simplification.